haystack/modeling/model/optimization.py

```diff
@@ -300,7 +294,7 @@ def optimize_model(
     local_rank: int,
     optimizer=None,
     distributed: Optional[bool] = False,
-    use_amp: Optional[str] = None,
+    use_amp: bool = False,
```

Comment:
This parameter is no longer needed. Is it safe for me to remove it? Or do I need to add some deprecation warning?

Reply:
This sounds good to me! I'll make the changes by adding the warning and the mapping.